module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/DefaultRecordTest.php

```diff
@@ -309,6 +310,7 @@ public function testGetOpenUrl()
             'rft.series' => 'Vico, Giambattista, 1668-1744. Works. 1982 ;',
             'rft.au' => 'Vico, Giambattista, 1668-1744.',
             'rft.pub' => 'Centro di Studi Vichiani,',
+            'rft.place' => 'Morano :',
```

Comment:
Blame the MARC21 standard, which puts trailing punctuation in a lot of fields, undermining the "machine-readable" part of MARC. Not my favorite thing! Maybe it's worth adding a regex to trim trailing punctuation and whitespace when adding the value to the OpenURL. Something like `preg_replace('/[:,\s]*/', '', $placesOfPublication[0])` would probably do the trick for 99% of cases (at least for records cataloged by American standards).

Reply:
Yeah it would make the OpenURL look better but probably break Zotero, if Zotero is assuming that the data already has the MARC21 punctuation and doesn't have to add its own.  I mean MLA, Chicago, APA probably don't all use the same formatting and it probably already has to remove the commas & colons for some of them, but Zotero is mature so I hope it knows what it's doing.  And it must expect this stupid trailing punctuation since as you say it's standard.

Reply:
But fair point about consistency. :-)

Reply:
I'd also say if we did the stripping (in a separate PR) it should be done in SolrMarc or some trait, not DefaultRecord, because who knows if ending whitespace or punctuation might actually be valid for some records.  I hate this stuff :)

Reply:
It would have to be done in a trait, because some of this data is pulled from the raw MARC record stored in Solr, not from the separate fields extracted by SolrMarc.